include/aws/http/connection_manager.h

```diff
@@ -25,6 +25,14 @@ typedef void(aws_http_connection_manager_on_connection_setup_fn)(
 
 typedef void(aws_http_connection_manager_shutdown_complete_fn)(void *user_data);
 
+struct aws_http_manager_metric {
+    /* The number of additional concurrent requests that can be supported by the HTTP manager without needing to
+     * establish additional connections to the target server. */
+    size_t available_concurrency;
```

Comment:
ask java team what they want here:

"The number of additional concurrent requests that can be supported by the HTTP client without needing to establish"

or

"For HTTP/2 operations, this is equal to the number of streams that are currently idle/unused."

because we make new connections when an "ideal" number is reached, not just when a "max" number is reached